### PR TITLE
Added I AM auth for Opensearch

### PR DESCRIPTION
**Pull request by syedzoherer**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [X] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by syedzoherer** [comment]
To access Opensearch with IAM auth, request needs to be signed with IAM auth. This change will allow users requests to be signed if OPENSEARCH_USE_AWS_IAM_AUTH is set to true.

**Review by RyanHolstien — CHANGES_REQUESTED** [review]

File: metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java [hunk 1]
```diff
@@ -65,9 +74,9 @@ protected RestHighLevelClient createInstance() {
     RestClientBuilder restClientBuilder;
     if (useSSL) {
       restClientBuilder = loadRestHttpsClient(host, port, pathPrefix, threadCount, connectionRequestTimeout, sslContext, username,
-          password);
-    } else if (username != null && password != null) {
-      restClientBuilder = loadRestHttpClient(host, port, pathPrefix, threadCount, connectionRequestTimeout, username, password);
+          password, opensearchUseAwsIamAuth, region);
+    } else if ((username != null && password != null) || opensearchUseAwsIamAuth) {
```
[the remark this answers is not in the document]
Comment (syedzoherer) [reply metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java]: Code refactor and removed default Region.

File: metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java [hunk 2]
```diff
@@ -91,30 +98,36 @@ private static RestClientBuilder loadRestHttpClient(@Nonnull String host, int po
...
-        credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(username, password));
-        httpAsyncClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
-        
+
+        if (username != null && password != null) {
+          final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+          credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(username, password));
+          httpAsyncClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+        }
+        if(opensearchUseAwsIamAuth) {
```
Comment (RyanHolstien) [inline metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java:116]: Fix checkstyle, needs a space after if

Comment (syedzoherer) [reply metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java]: Fixed.

**Review by RyanHolstien — APPROVED** [review]

**Pull request merged**